src/analytics/create-step-tracker.ts

```diff
@@ -0,0 +1,55 @@
+import localStorageFallback from 'local-storage-fallback';
+
+import { CheckoutService } from '../checkout';
+import { MissingDataError, MissingDataErrorType } from '../common/error/errors';
+
+import AnalyticsStepTracker, { StepTrackerConfig } from './analytics-step-tracker';
+import { isAnalyticsTrackerWindow } from './is-analytics-step-tracker-window';
+import NoopStepTracker from './noop-step-tracker';
+import StepTracker from './step-tracker';
+
+/**
+ * Creates an instance of `StepTracker`.
+ *
+ * @remarks
+ * ```js
+ * const checkoutService = createCheckoutService();
+ * await checkoutService.loadCheckout();
+ * const stepTracker = createStepTracker(checkoutService, { shouldBackfillPreviousSteps: true });
```

Comment:
potentially the config options could only have `checkoutSteps: [ ... ]`, if not passed it will assume the default, if passed empty won't backfill.